platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -574,135 +735,178 @@ std::optional<std::pair<Response, uint64_t>> OfflineDatabase::getTile(const Reso
         size = data->length();
     }
 
-    return std::make_pair(response, size);
+    return std::make_pair(response, size); // Return the response and data size
 }
 
 std::optional<int64_t> OfflineDatabase::hasTile(const Resource::TileData& tile) {
-    // clang-format off
-    mapbox::sqlite::Query size{ getStatement(
-        "SELECT length(data) "
-        "FROM tiles "
-        "WHERE url_template = ?1 "
-        "  AND pixel_ratio  = ?2 "
-        "  AND x            = ?3 "
-        "  AND y            = ?4 "
-        "  AND z            = ?5 ") };
-    // clang-format on
+    // First, try to find the tile in the 'tiles' table
+    std::optional<int64_t> selectTilesResult = extractTileDataSize(
+        tile,
+        "SELECT length(data) FROM tiles WHERE url_template = ?1 AND pixel_ratio = ?2 AND x = ?3 AND y = ?4 AND z = ?5");
+    if (selectTilesResult) {
+        // std::cout << "-------- HASTILE - FOUND IN TILES\n";
+        return selectTilesResult;
+    } else {
+        // If not found in 'tiles', try the 'ambient_tiles' table
+        std::optional<int64_t> selectAmbientTilesResult = extractTileDataSize(
+            tile,
+            "SELECT length(data) FROM ambient_tiles WHERE url_template = ?1 AND pixel_ratio = ?2 AND x = ?3 AND y = ?4 "
+            "AND z = ?5");
+        if (selectAmbientTilesResult) {
+            // std::cout << "-------- HASTILE - FOUND IN AMBIENT_TILES\n";
```

Comment:
There are commented out lines that print used for debugging throughout this file.

Please remove them or change them to (debug) logging.